cstool/cstool_tricore.c

```diff
@@ -44,29 +44,43 @@ void print_insn_detail_tricore(csh handle, cs_insn *ins)
 			break;
 		}
 
-		// Print out all registers accessed by this instruction (either implicit or
-		// explicit)
-		if (!cs_regs_access(handle, ins, regs_read, &regs_read_count,
-				    regs_write, &regs_write_count)) {
-			if (regs_read_count) {
-				printf("\tRegisters read:");
-				for (i = 0; i < regs_read_count; i++) {
-					printf(" %s",
-					       cs_reg_name(handle,
-							   regs_read[i]));
-				}
-				printf("\n");
+		switch (op->access) {
+		default:
+			break;
+		case CS_AC_READ:
+			printf("\t\toperands[%u].access: READ\n", i);
+			break;
+		case CS_AC_WRITE:
+			printf("\t\toperands[%u].access: WRITE\n", i);
+			break;
+		case CS_AC_READ | CS_AC_WRITE:
+			printf("\t\toperands[%u].access: READ | WRITE\n", i);
+			break;
+		}
+	}
+	// Print out all registers accessed by this instruction (either implicit or
+	// explicit)
+	if (!cs_regs_access(handle, ins, regs_read, &regs_read_count,
+			    regs_write, &regs_write_count)) {
+		if (regs_read_count) {
+			printf("\tRegisters read:");
+			for (i = 0; i < regs_read_count; i++) {
```

Comment:
Couldn't you just iterate over `ins->detail->tricore->operands` here and don't call `cs_regs_access()`? Or am I missing something?

Reply:
Although this is copied from `cstool/cstool_arm.c`.
I think `ins->detail->tricore->operands` and `ins->detail->{regs_write,regs_read}` are supposed to be a little different? Although most of them are duplicated, `ins->detail->{regs_write,regs_read}` may contain implicit register read/write information?

Reply:
Yeah. You are right. And I just saw that I forgot to change this in the ARM PR.
Ignore it then.